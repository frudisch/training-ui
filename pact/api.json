{
  "provider": {
    "name": "training-service"
  },
  "consumer": {
    "name": "training-ui"
  },
  "interactions": [
    {
      "description": "Selecting all events / trainings saved in the backend",
      "request": {
        "method": "GET",
        "path": "/"
      },
      "response": {
        "status": 200,
        "headers": {
          "Content-Type": "application/json"
        },
        "body": [{
          "date": "03.01.2023",
          "end": "20:00",
          "start": "18:00"
        }, {
          "date": "07.01.2023",
          "end": "20:00",
          "start": "18:00",
          "enemy": "FÃ¼chse Berlin"
        }]
      },
      "providerStates": [
        {
          "name": "test GET"
        }
      ]
    }
  ],
  "metadata": {
    "pact-specification": {
      "version": "3.0.0"
    },
    "pact-jvm": {
      "version": "3.6.3"
    }
  }
}