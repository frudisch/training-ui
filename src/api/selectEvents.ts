import {EventTyp} from "../event-list/EventList";
import axios from "axios";
import configuration from "../configuration";

export default async (): Promise<EventTyp[]> => {
    const response = await axios.get<EventTyp[]>(`${configuration.backendURL}/event/`);
    return response.data;
}