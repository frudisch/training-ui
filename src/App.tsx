import React, {useEffect, useState} from 'react';
import './App.css';
import EventList, {EventTyp} from "./event-list/EventList";
import selectEvents from "./api/selectEvents";

const App = () => {
    const [events, setEvents] = useState<EventTyp[]>();
    useEffect(() => {
        const getEvents = async () => {
            const selectedEvents = await selectEvents();
            console.log(selectedEvents);
            setEvents(selectedEvents);
        }

        if(!events) {
            getEvents();
        }
    }, [events]);
    return (
        <div className="App">
            <header className="App-header">
                <EventList eventList={events}/>
            </header>
        </div>
    );
}

export default App;
