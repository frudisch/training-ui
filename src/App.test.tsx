import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';
import * as selectEvents from "./api/selectEvents";

jest.mock('./api/selectEvents');
const selectEventsMock = selectEvents as jest.Mocked<typeof selectEvents>;
describe('App', () => {
    it('should be rendered and match the snapshot', async () => {
        const expectedValues = [{
            date: '03.01.2023',
            end: '20:00',
            start: '18:00',
        }, {
            date: '07.01.2023',
            end: '20:00',
            start: '18:00',
            enemy: 'FÃ¼chse Berlin'
        }];
        selectEventsMock.default.mockResolvedValue(expectedValues);

        const rendered = renderer
            .create(<App></App>)
            .toJSON();

        expect(rendered).toMatchSnapshot();
    });
})
