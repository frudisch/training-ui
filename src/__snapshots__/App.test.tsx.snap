// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should be rendered and match the snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <div />
  </header>
</div>
`;
